metricbeat/helper/metricset.go

```diff
@@ -51,11 +70,34 @@ func (m *MetricSet) Fetch() error {
 
 		go func(h string) {
 
+			m.incrementFetcher()
+			defer m.decrementFetcher()
+
+			fetchCounter := atomic.LoadInt32(&m.fetchCounter)
```

Comment:
I recommend returning the incremented value from `incrementFetcher` and then using that value rather than loading it again (where it could theoretically have been changed since you incremented).